src/runtime_src/core/common/query_requests.cpp

```diff
@@ -28,46 +27,42 @@ to_string(xrt_core::query::p2p_config::value_type value)
   return p2p_config_map[value];
 }
 
+std::map<std::string, int64_t>
+xrt_core::query::p2p_config::
+to_map(const xrt_core::query::p2p_config::result_type& config)
+{
+  std::map<std::string, int64_t> config_map;
+  for (auto& str : config) {
+    // str is in key:value format obtained from p2p_config query
+    const auto pos = str.find(":");
+    const std::string config_item = str.substr(0, pos);
+    try {
+      const int64_t value = std::stoll(str.substr(pos + 1));
+      config_map[config_item] = value;
+    } catch (const std::exception&) {
+      // Failed to parse a non long long BAR value. Dont parse it into the map and move on!
+    }
+  }
+  return config_map;
+}
+
 std::pair<xrt_core::query::p2p_config::value_type, std::string>
 xrt_core::query::p2p_config::
 parse(const xrt_core::query::p2p_config::result_type& config)
 {
-  int64_t bar = -1;
-  int64_t rbar = -1;
-  int64_t remap = -1;
-  int64_t exp_bar = -1;
-
-  // parse the query
-  for(const auto& val : config) {
-    auto pos = val.find(':') + 1;
-    if(val.find("rbar") == 0)
-      rbar = std::stoll(val.substr(pos));
-    else if(val.find("exp_bar") == 0)
-      exp_bar = std::stoll(val.substr(pos));
-    else if(val.find("bar") == 0)
-      bar = std::stoll(val.substr(pos));
-    else if(val.find("remap") == 0)
-      remap = std::stoll(val.substr(pos));
-  }
+  auto config_map = xrt_core::query::p2p_config::to_map(config);
 
   // return the config with a message
-  if (bar == -1) {
-    return {xrt_core::query::p2p_config::value_type::not_supported,
-            "P2P config failed. P2P is not supported. Can't find P2P BAR."};
-  }
-  else if (rbar != -1 && rbar > bar) {
-    return {xrt_core::query::p2p_config::value_type::reboot, 
-            "Warning:Please WARM reboot to enable p2p now."};
-  }
-  else if (remap > 0 && remap != bar) {
-    return {xrt_core::query::p2p_config::value_type::error,
-            "Error:P2P config failed. P2P remapper is not set correctly"};
-  }
-  else if (bar == exp_bar) {
+  if (config_map.find("bar") == config_map.end())
```

Comment:
This nested if loop is not needed for if the condition matches it returns.

Please remove replace the "else" with a new line.